tests/test_book_routes.py

```diff
@@ -28,6 +92,43 @@ def test_create_one_book(client):
     })
     response_body = response.get_json()
 
+    # Assert
+    assert response.status_code == 201
+    assert response_body == {
+        "id": 1,
+        "title": "New Book",
+        "description": "The Best!"
+    }
+
+def test_create_one_book_no_title(client):
+    # Arrange
+    test_data = {"description": "The Best!"}
+
+    # Act & Assert
+    with pytest.raises(KeyError, match="title"):
+        response = client.post("/books", json=test_data)
+
+def test_create_one_book_no_description(client):
+    # Arrange
+    test_data = {"title": "New Book"}
+
+    # Act & Assert
+    with pytest.raises(KeyError, match="description"):
+        response = client.post("/books", json=test_data)
+
+def test_create_one_book_with_extra_keys(client):
+    # Arrange
+    test_data = {
+        "extra": "some stuff",
+        "title": "New Book",
+        "description": "The Best!",
+        "another": "last value"
+    }
+
+    # Act
+    response = client.post("/books", json=test_data)
+    response_body = response.get_json()
+
```

Comment:
Shouldn't these not be added until 07b? Learn introduces them along with the from_dict refactor: https://github.com/Ada-Developers-Academy/core-unit-2/blob/main/api-7-refactoring/extracting-helper-functions-from-dict.md

Reply:
This came about because the create route now returns a Book record instead of a string. The create route is a dependency for the `to_dict` refactor, which is why I moved these tests up.

Reply:
Ah, I see. Would it be strange to only add the successful test cases here (the ones that actually make it through to returning the dict)? Per my other comment, having the `KeyError` tests is a little odd here. And if we're going to immediately turn around and fix them for the message output, it seems unnecessary to get into those weeds here.

Reply:
I don't feel that it's a great example to purposefully leave off edge cases when we're making changes that could be affected by those edge cases. My thoughts are that a project's test suite has to evolve with the code over time, and it isn't a bad thing to show off that sometimes we do need to revisit tests because we made a change to the logic to make our app stronger overall. 

In order to keep our lessons focused, I don't really want to add changes to code that isn't otherwise handled in this lesson. The next lesson handles Book creation, so it makes sense to me to leave error handling for the Book creation bundled with that lesson. I'd want to get other folk's thoughts, but if you think it's a detriment to students, what do you think about swapping the order of the lessons so the `from_dict` refactor comes first?

Reply:
Oh, I like that. We can implement the `from_dict`, but still leave the existing repetative return value, then fix the `to_dict` in the next section. That would allow us to skip the `KeyError` issue pretty neatly.